cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/MacroParser.java

```diff
@@ -119,7 +136,11 @@ private MacroMetadata findRightmostMacro(String str)
       return getMacroFunctionMetadata(startIndex, endIndex, macroStr, argsStartIndex);
     } else {
       macroStr = replaceEscapedSyntax(macroStr);
-      return new MacroMetadata(macroEvaluator.lookup(macroStr), startIndex, endIndex);
+      if (lookupsEnabled) {
+        return new MacroMetadata(macroEvaluator.lookup(macroStr), startIndex, endIndex, true);
+      } else {
+        return new MacroMetadata(String.format("${%s}", macroStr), startIndex, endIndex, false);
```

Comment:
can you add a comment something around, `when lookups is disabled, we create macro string and make recursive evaluation false, so this string wont be evaluated again`